src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -437,14 +437,26 @@ let pp_ctor ppf p =
 let rec top_level_decls Stmt.Fixed.({pattern; _}) =
   match pattern with
   | Decl d when d.decl_id <> "pos__" ->
-      [(d.decl_id, Type.to_unsized d.decl_type, UnsizedType.DataOnly)]
+      [(d.decl_id, Type.to_unsized d.decl_type)]
   | SList stmts -> List.concat_map ~f:top_level_decls stmts
   | _ -> []
 
 (** Print the private data members of the model class *)
 let pp_model_private ppf {Program.prepare_data; _} =
   let data_decls = List.concat_map ~f:top_level_decls prepare_data in
-  pf ppf "%a" (list ~sep:cut pp_decl) data_decls
+  (*Filter out Any data that is not an Eigen matrix*)
+  let get_eigen_map (name, ut) =
+    match (UnsizedType.is_eigen_type ut, Transform_Mir.is_opencl_var name) with
+    | true, false -> Some (name, ut)
+    | false, _ -> None
+    | true, _ -> None
```

Comment:
Totally cool with this change, though one thing I don't get yet. When is it better to use an if vs. a match? Is it just a style choice? I feel like when I see ifs I usually see us do a `;` at the end I get kind of confused about execution stuff so I'm like "Oh idt I like that" but are ifs totally fine?

Reply:
I dunno, I think it's just a style choice. Personally I prefer `if`/`when` for booleans and `match` for anything more complex. Also `if` can omit `else` so it's a bit more compact if you only need one branch.

If you want to keep `match` maybe merge `| true, _ -> None | false, _ -> None` into a single `| _, _ -> None`?
Actually, thinking more about this, it's more natural to use `List.filter` here instead of `List.filter_map`.

Reply:
Makes sense! Switched this to use `filter` and `if`